app/models/user.rb

```diff
@@ -26,12 +27,33 @@ def remember
     update_attribute(:remember_digest, User.digest(remember_token))
   end
 
-  def authenticated? remember_token
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  def authenticated?(attribute, token)
+    digest = send("#{attribute}_digest")
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password?(token)
   end
 
   def forget
     update_attribute(:remember_digest, nil)
   end
+
+  def activate
+    update_attribute(:activate, true)
+    update_attribute(:activate_at, Time.zone.now)
+  end
+
+  def send_activation_email
+    UserMailer.account_activation(self).deliver_now
+  end
+
+  private
+
+  def downcase_email
+    self.email = email.downcase
```

Comment:
so sánh cho chị `downcase` và `downcase!` có gì khác nhau. chị dùng có ! dc ko

Reply:
.

Reply:
Trong trường hợp này là không được chị vì dowcase!  sẽ trả về nill nếu như chuỗi đó toàn chữ thường chị, như vậy sẽ sai trong trường hợp này . Dùng dowcase thì nếu toàn chữ thường thì trả về chính nó chị.

Reply:
ukm